src/main/java/org/prebid/server/bidder/yieldone/YieldoneBidder.java

```diff
@@ -0,0 +1,159 @@
+import org.prebid.server.util.HttpUtil;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.stream.Collectors;
+
+/**
+ * Yieldone {@link Bidder} implementation.
+ */
+public class YieldoneBidder implements Bidder<BidRequest> {
+    private static final TypeReference<ExtPrebid<?, ExtImpYieldone>> YIELDONE_EXT_TYPE_REFERENCE =
+            new TypeReference<ExtPrebid<?, ExtImpYieldone>>() {
+            };
+    private static final String DEFAULT_BID_CURRENCY = "USD";
+    private final String endpointUrl;
+    private final JacksonMapper mapper;
+
+    public YieldoneBidder(String endpointUrl, JacksonMapper mapper) {
+        this.endpointUrl = HttpUtil.validateUrl(Objects.requireNonNull(endpointUrl));
+        this.mapper = Objects.requireNonNull(mapper);
+    }
+
+    @Override
+    public Result<List<HttpRequest<BidRequest>>> makeHttpRequests(BidRequest request) {
+        final List<BidderError> errors = new ArrayList<>();
+        final List<Imp> validImps = new ArrayList<>();
+
+        for (Imp imp : request.getImp()) {
+            try {
+                final Imp validImp = validateImp(imp);
+                final ExtImpYieldone extImp = parseImpExt(imp);
+
+                validImps.add(validImp);
+
+            } catch (PreBidException e) {
+                errors.add(BidderError.badInput(e.getMessage()));
+            }
+        }
+
+        final BidRequest outgoingRequest = request.toBuilder().imp(validImps).build();
+        final String body = mapper.encode(outgoingRequest);
+
+        return Result.of(Collections.singletonList(
+                HttpRequest.<BidRequest>builder()
+                        .method(HttpMethod.POST)
+                        .uri(endpointUrl)
+                        .headers(HttpUtil.headers())
+                        .payload(outgoingRequest)
+                        .body(body)
+                        .build()),
+                errors);
+    }
+
+    private Imp validateImp(Imp imp) {
+        final Banner banner = imp.getBanner();
+        if (banner != null) {
+            if (banner.getH() == null && banner.getW() == null && banner.getFormat().size() > 0) {
+                final Format firstFormat = banner.getFormat().get(0);
+                final Banner modifiedBanner = banner.toBuilder()
+                        .h(firstFormat.getH())
+                        .w(firstFormat.getW())
+                        .build();
+                return imp.toBuilder().banner(modifiedBanner).build();
+            }
+        }
+        return imp;
+    }
+
+    private ExtImpYieldone parseImpExt(Imp imp) {
+        try {
+            return mapper.mapper().convertValue(imp.getExt(), YIELDONE_EXT_TYPE_REFERENCE).getBidder();
+        } catch (IllegalArgumentException e) {
+            throw new PreBidException(e.getMessage(), e);
+        }
+    }
```

Comment:
This method is not necessary. 
We validating all required (that was specified in .json) parameters in auction workflow. 
Buy you can change it to test ext. 
`validateImpExt` for example (which will return void)